runtime/interpreter/interpreter.go

```diff
@@ -4733,3 +4739,35 @@ func (interpreter *Interpreter) DecodeTypeInfo(decoder *cbor.StreamDecoder) (atr
 func (interpreter *Interpreter) Storage() Storage {
 	return interpreter.SharedState.Config.Storage
 }
+
+// ConfigureAccountLinkingAllowed configures if execution is allowed to use account linking,
+// depending on the occurrence of the pragma declaration #allowAccountLinking.
+//
+// The pragma declaration must appear as a top-level declaration (i.e. not nested in the program),
+// and must appear before all other declarations (i.e. at the top of the program).
+//
+// This requirement is also checked statically.
+//
+// This is a temporary feature, which is planned to get replaced by capability controllers,
+// and a new Account type with entitlements.
+func (interpreter *Interpreter) ConfigureAccountLinkingAllowed() {
+	config := interpreter.SharedState.Config
+
+	config.AccountLinkingAllowed = false
+
+	var rejectAllowAccountLinkingPragma bool
+
+	for _, declaration := range interpreter.Program.Program.Declarations() {
+
+		if pragmaDeclaration, isPragma := declaration.(*ast.PragmaDeclaration); isPragma {
+			if sema.IsAllowAccountLinkingPragma(pragmaDeclaration) {
+				if !rejectAllowAccountLinkingPragma {
+					config.AccountLinkingAllowed = true
```

Comment:
It seems the loop can be terminated on the first occurrence of the pragma. That would also allow removing the `rejectAllowAccountLinkingPragma` variable.

Reply:
Good idea! I had pretty much just copied the logic from the checker, but he loop is unnecessary.

In [`c5a4e71` (#2380)](https://github.com/onflow/cadence/pull/2380/commits/c5a4e71957c99f4b3a7071cd044df078bcc62114) I made the function as strict as the checker, and simplified it, by just looking at the first pragma. In addition, I added some more checker and interpreter tests.